NullPointerException attempting to inject DataSource without jdbc feature enabled

When an application attempts to inject (@Resource) a DataSource, but a JDBC feature is not enabled in server.xml, then the application fails to start with a NullPointerException in the InjectionEngine as follows:

```
Stack Dump = com.ibm.wsspi.injectionengine.InjectionException: java.lang.NullPointerException
	at com.ibm.wsspi.injectionengine.InjectionBinding.createDefinitionReference(InjectionBinding.java:1241)
	at com.ibm.ws.injectionengine.processor.DataSourceDefinitionInjectionBinding.resolve(DataSourceDefinitionInjectionBinding.java:506)
	at com.ibm.ws.injectionengine.processor.DataSourceDefinitionProcessor.resolve(DataSourceDefinitionProcessor.java:79)
	at com.ibm.wsspi.injectionengine.InjectionProcessor.resolveInjectionBindings(InjectionProcessor.java:631)
	at com.ibm.wsspi.injectionengine.InjectionProcessorContextImpl.processBindings(InjectionProcessorContextImpl.java:59)
	at com.ibm.ws.injectionengine.InjectionProcessorManager.processBindings(InjectionProcessorManager.java:510)
	at com.ibm.ws.injectionengine.AbstractInjectionEngine.processInjectionMetaData(AbstractInjectionEngine.java:484)
	at com.ibm.ws.injectionengine.osgi.internal.OSGiInjectionEngineImpl.processInjectionMetaDataPrivileged(OSGiInjectionEngineImpl.java:305)
	at com.ibm.ws.injectionengine.osgi.internal.OSGiInjectionEngineImpl.access$000(OSGiInjectionEngineImpl.java:114)
	at com.ibm.ws.injectionengine.osgi.internal.OSGiInjectionEngineImpl$1.run(OSGiInjectionEngineImpl.java:247)
	at com.ibm.ws.injectionengine.osgi.internal.OSGiInjectionEngineImpl$1.run(OSGiInjectionEngineImpl.java:244)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:734)
	at com.ibm.ws.injectionengine.osgi.internal.OSGiInjectionEngineImpl.processInjectionMetaData(OSGiInjectionEngineImpl.java:244)
	at com.ibm.ws.injectionengine.ReferenceContextImpl.processImpl(ReferenceContextImpl.java:588)
	at com.ibm.ws.injectionengine.ReferenceContextImpl.process(ReferenceContextImpl.java:287)
	at com.ibm.ws.injectionengine.osgi.internal.OSGiReferenceContextImpl.process(OSGiReferenceContextImpl.java:30)
	at com.ibm.ws.webcontainer.osgi.webapp.WebApp.commonInitializationStart(WebApp.java:252)
	at com.ibm.ws.webcontainer.webapp.WebApp.initialize(WebApp.java:1014)
	at com.ibm.ws.webcontainer.webapp.WebApp.initialize(WebApp.java:6675)
	at com.ibm.ws.webcontainer.osgi.DynamicVirtualHost.startWebApp(DynamicVirtualHost.java:470)
	at com.ibm.ws.webcontainer.osgi.DynamicVirtualHost.startWebApplication(DynamicVirtualHost.java:465)
	at com.ibm.ws.webcontainer.osgi.WebContainer.startWebApplication(WebContainer.java:1179)
	at com.ibm.ws.webcontainer.osgi.WebContainer.access$100(WebContainer.java:110)
	at com.ibm.ws.webcontainer.osgi.WebContainer$3.run(WebContainer.java:976)
	at com.ibm.ws.threading.internal.ExecutorServiceImpl$RunnableWrapper.run(ExecutorServiceImpl.java:238)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.NullPointerException
	at com.ibm.ws.injectionengine.osgi.internal.OSGiInjectionEngineImpl.getResourceFactoryBuilder(OSGiInjectionEngineImpl.java:580)
	at com.ibm.ws.injectionengine.osgi.internal.OSGiInjectionEngineImpl.createDefinitionResourceFactoryReference(OSGiInjectionEngineImpl.java:598)
	at com.ibm.ws.injectionengine.osgi.internal.OSGiInjectionEngineImpl.createDefinitionReference(OSGiInjectionEngineImpl.java:543)
	at com.ibm.wsspi.injectionengine.InjectionBinding.createDefinitionReference(InjectionBinding.java:1237)
	... 29 more

```

The NullPointerException should be handled and a more meaningful error logged, such as:
```
CWNEN0011E: The injection engine failed to process bindings for the metadata due to the following error: com.ibm.wsspi.injectionengine.InjectionException: javax.sql.DataSource definitions are not supported in this server configuration 
```

--- a/dev/com.ibm.ws.injection/src/com/ibm/ws/injectionengine/osgi/internal/OSGiInjectionEngineImpl.java
+++ b/dev/com.ibm.ws.injection/src/com/ibm/ws/injectionengine/osgi/internal/OSGiInjectionEngineImpl.java
@@ -577,7 +577,7 @@ public class OSGiInjectionEngineImpl extends AbstractInjectionEngine implements 
     @Override
     public ResourceFactoryBuilder getResourceFactoryBuilder(String type) throws InjectionException {
         Iterator<ResourceFactoryBuilder> builderIter = resourceFactoryBuilders.getServices(type);
-        ResourceFactoryBuilder builder = builderIter.hasNext() ? builderIter.next() : null;
+        ResourceFactoryBuilder builder = builderIter != null && builderIter.hasNext() ? builderIter.next() : null;
         if (builder == null) {
             throw new InjectionException(type + " definitions are not supported in this server configuration");
         }

